R/Canada.R

```diff
@@ -0,0 +1,78 @@
+#' Canada Class containing country specific attributes and methods
+#'
+#' @description Country specific information for downloading, cleaning
+#'  and processing covid-19 region data for Canada.
+#'
+#' Data is sourced from \url{https://health-infobase.canada.ca}.
+#' @details Inherits from `DataClass`
+#' @source https://health-infobase.canada.ca
+#' @export
+#' @examples
+#' \dontrun{
+#' region <- Canada$new(verbose = TRUE, steps = TRUE)
+#' region$download()
+#' region$clean()
+#' region$process()
+#' region$return()
+#' }
+Canada <- R6::R6Class("Canada",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field level_1_region the level 1 region name.
+    level_1_region = "level_1_region", # for brevity - refers to
+    # provinces and territories
+    #' @field data_url link to raw data
+    data_url = "https://health-infobase.canada.ca/src/data/covidLive/covid19.csv", #nolint
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "cases_total", "deaths_new", "recovered_total", "tested_new"),
+
+    #' @description *Canada* specific provincial/territorial level data
+    #' cleaning
+    #' @param ... pass additional arguments
+    #'
+    #' @importFrom dplyr filter select mutate rename
+    #' @importFrom tidyr replace_na
+    #' @importFrom lubridate dmy
+    clean = function(...) {
+      # function to clean the data (MUST BE CALLED clean)
+      # modify the data variable 'region' in place and add using 'self'
+      # e.g. self$data$clean <- something
+      # No return statement is required
+      # have a statement like this to indicate information to user if requested
+      if (self$verbose) {
```

Comment:
If possible port in message_verbose. No problem if not around as I can make these updates prior to merging.

Reply:
Just doing this.